src/paradigma/heart_rate/heart_rate_analysis.py

```diff
@@ -29,3 +35,55 @@ def extract_signal_quality_features_io(input_path: Union[str, Path], output_path
     # Extract gait features
     df_windowed = extract_signal_quality_features(df, config)
     return df_windowed
+
+
+def signal_quality_classification(df: pd.DataFrame, config: SignalQualityClassificationConfig, path_to_classifier_input: Union[str, Path]) -> pd.DataFrame:
+    """
+    Classify the signal quality of the PPG signal using a logistic regression classifier.
+    The classifier is trained on features extracted from the PPG signal.
+    The features are extracted using the extract_signal_quality_features function.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The DataFrame containing the PPG signal.
+    config : SignalQualityClassificationConfig
+        The configuration for the signal quality classification.
+    path_to_classifier_input : Union[str, Path]
+        The path to the directory containing the classifier.
+
+    Returns
+    -------
+    pd.DataFrame
+        The DataFrame containing the PPG signal quality predictions.
+    """
+    
+    clf = pd.read_pickle(os.path.join(path_to_classifier_input, 'classifiers', config.classifier_file_name))
+    lr_clf = clf['model']
+    mu = clf['mu']
+    sigma = clf['sigma']
+
+    # Prepare the data
+    lr_clf.feature_names_in_ = ['var', 'mean', 'median', 'kurtosis', 'skewness', 'f_dom', 'rel_power', 'spectral_entropy', 'signal_to_noise', 'auto_corr']
```

Comment:
Je zou nog kunnen kijken of je deze feature names bij het wegschrijven van de classifier naar pickle mee kan geven. Dan definieer je ze niet twee keer, en kan het ook niet mis gaan als je er één aanpast.